src/xzdec/meson.build

```diff
@@ -0,0 +1,89 @@
+## SPDX-License-Identifier: 0BSD
+## Author: Vincent Torri
+
+if get_option('xzdec')
+    xzdec_src = [
+        '../common/tuklib_exit.c',
+        '../common/tuklib_progname.c',
+        'xzdec.c'
+    ]
+
+    xzdec_cargs = [
+        '-DHAVE_CONFIG_H',
+        '-DHAVE_STDBOOL_H',
+        '-DHAVE_INTTYPES_H',
+        '-DHAVE_STDINT_H',
+        '-DTUKLIB_GETTEXT=0'
+    ]
```

Comment:
I suggest puting `HAVE_STDBOOL_H`, `HAVE_INTTYPES_H`, and `HAVE_STDINT_H` into config.h in the top-level meson.build. The Meson build won't run on systems that lack them. After 5.8.0, XZ Utils will assume those to be present as well, making the macros unneeded.

Similarly, perhaps `HAVE_CONFIG_H` could be set at the top-level file with `add_project_arguments('-DHAVE_CONFIG_H')`.

`-DTUKLIB_GETTEXT=0` has to be kept here local to `xzdec` and `lzmadec`.

Reply:
I move them at the top level meson.build for the whole project ( in f51a11f845a064f8410dd1cb3411dfd0344d0dad)